rfc/0016-instances-vs-types/0016-instances-vs-types.md

```diff
@@ -0,0 +1,248 @@
+##### The model
+```
+pipeline CarsPipeline {
+	CarsExtractor
+	   	-> CarsTableInterpreter
+		-> CarsLoader;
+
+	block CarsCSVExtractor oftype CSVExtractor {
+		url: "https://gist.githubusercontent.com/noamross/e5d3e859aa0c794be10b/raw/b999fb4425b54c63cab088c0ce2c0d6ce961a563/cars.csv";
+	}
+
+	block CarsTableInterpreter oftype TableInterpreter {
+		header: true;
+		columns: [
+			"name" oftype text,
+			"mpg" oftype decimal,
+		];
+	}
+
+	block CarsLoader oftype SQLiteLoader {
+		table: "Cars";
+		file: "./cars.sqlite";
+	}
+}
+```
+
+##### As diagram including M2 level
+```mermaid
+flowchart BT
+subgraph M2
+    Pipeline
+    Block
+    Blocktype
+    
+    BuiltInBlocktype
+    CompositeBlocktype
+end
+subgraph M1
+    CarsPipeline
+    CSVExtractor
+    TableInterpreter
+    SQLiteLoader
+    CarsCSVExtractor
+    CarsTableInterpreter
+    CarsLoader
+end
+subgraph M0
+    anonymousPipeline1["anonymous"]
+    
+    anonymousBlock1["anonymous"]
+    anonymousBlock2["anonymous"]
+    anonymousBlock3["anonymous"]
+end
+
+BuiltInBlocktype -."is subtype of".-> Blocktype
+CompositeBlocktype -."is subtype of".-> Blocktype
+
+CarsPipeline --"instance of"--> Pipeline
+CSVExtractor --"instance of"--> CompositeBlocktype
+TableInterpreter --"instance of"--> BuiltInBlocktype
+SQLiteLoader --"instance of"--> BuiltInBlocktype
+CarsCSVExtractor --"instance of"--> Block
+CarsTableInterpreter --"instance of"--> Block
+CarsLoader --"instance of"--> Block
+
+CarsCSVExtractor -."oftype".-> CSVExtractor
+CarsTableInterpreter -."oftype".-> TableInterpreter
+CarsLoader -."oftype".-> SQLiteLoader
+
+anonymousPipeline1 --"instance of"--> CarsPipeline
+
+anonymousBlock1 --"instance of"--> CarsCSVExtractor
+anonymousBlock2 --"instance of"--> CarsTableInterpreter
+anonymousBlock3 --"instance of"--> CarsLoader
+```
+
+##### As diagram excluding M2 level/instance of relationships
+```mermaid
+flowchart BT
+subgraph M1
+    direction BT
+    CarsPipeline["CarsPipeline : Pipeline"]
+    CSVExtractor["CSVExtractor : CompositeBlocktype"]
+    TableInterpreter["TableInterpreter : BuiltInBlocktype"]
+    SQLiteLoader["SQLiteLoader : BuiltInBlocktype"]
+    CarsCSVExtractor["CarsCSVExtractor : Block"]
+    CarsTableInterpreter["CarsTableInterpreter : Block"]
+    CarsLoader["CarsTableInterpreter : Block"]
+end
+subgraph M0
+    anonymousPipeline1[" : CarsPipeline"]
+    
+    anonymousBlock1[" : CarsCSVExtractor"]
+    anonymousBlock2[" : CarsTableInterpreter"]
+    anonymousBlock3[" : CarsLoader"]
+end
+
+CarsCSVExtractor -."oftype : OftypeRelationship".-> CSVExtractor
+CarsTableInterpreter -."oftype : OftypeRelationship".-> TableInterpreter
+CarsLoader -."oftype : OftypeRelationship".-> SQLiteLoader
+M0 ~~~ M1
+```
+
+### Proposed changes
+#### 1. Clarify naming
+- Across M-levels, we have only a `instance of` relationship
+- On M1 level, only the `oftype` relationship exists between `block`s and `blocktype`s because it is modelled at M2 level
```

Comment:
```suggestion
- The only other currently possible relationship is the `oftype` relationship between a `block` and a `blocktype` on M1 level. Other relationships are not possible since they are not modelled at M2 level.
```